src/Illuminate/Database/Connectors/MySqlConnector.php

```diff
@@ -104,24 +104,51 @@ protected function getHostDsn(array $config)
     }
 
     /**
+     * Get the modes for the connection.
+     *
+     * @param  array  $config
+     * @return array
+     */
+    protected function getModes(array $config)
+    {
+        if (isset($config['modes'])) {
+            return (array) $config['modes'];
+        }
+
+        if (! isset($config['strict'])) {
+            return [];
+        }
+
+        if (! $config['strict']) {
+            return ['NO_ENGINE_SUBSTITUTION'];
+        }
+
+        return [
+            'ERROR_FOR_DIVISION_BY_ZERO',
+            'NO_AUTO_CREATE_USER',
+            'NO_ENGINE_SUBSTITUTION',
+            'NO_ZERO_DATE',
+            'NO_ZERO_IN_DATE',
+            'ONLY_FULL_GROUP_BY',
+            'STRICT_TRANS_TABLES',
+        ];
+    }
+
+    /**
      * Set the modes for the connection.
      *
      * @param  \PDO  $connection
      * @param  array  $config
-     * @return void
+     * @return bool
      */
     protected function setModes(PDO $connection, array $config)
     {
-        if (isset($config['modes'])) {
-            $modes = implode(',', $config['modes']);
-
-            $connection->prepare("set session sql_mode='".$modes."'")->execute();
-        } elseif (isset($config['strict'])) {
-            if ($config['strict']) {
-                $connection->prepare("set session sql_mode='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION'")->execute();
-            } else {
-                $connection->prepare("set session sql_mode='NO_ENGINE_SUBSTITUTION'")->execute();
-            }
+        if ($modes = $this->getModes($config)) {
```

Comment:
If `$config['strict']` is not set, `getModes` returns an empty array which is falsey and skips this condition. We definitely _do_ want it to be skipped, because if someone doesn't set `strict` or `modes`, we don't want to override their global database sql_mode with an empty sql_mode, but I don't think what we have right now expresses that very clearly.

Maybe we need to do something like:

```php
protected function setModes(PDO $connection, array $config)
{
    if ($this->shouldSetSqlMode($config)) {
        $sqlModes = str_replace("'", '', implode(',', $this->getModes($config)));
        $connection->prepare("set session sql_mode='".$sqlModes."'")->execute();
    }
}

protected function shouldSetSqlMode(array $config)
{
    return isset($config['modes']) || isset($config['strict']);
}

protected function getModes(array $config)
{
    if (isset($config['modes'])) {
        return (array) $config['modes'];
    }

    if ($config['strict'] === false) {
        return ['NO_ENGINE_SUBSTITUTION'];
    }

    if ($config['strict'] === true) {
        return [
            'ERROR_FOR_DIVISION_BY_ZERO',
            'NO_AUTO_CREATE_USER',
            'NO_ENGINE_SUBSTITUTION',
            'NO_ZERO_DATE',
            'NO_ZERO_IN_DATE',
            'ONLY_FULL_GROUP_BY',
            'STRICT_TRANS_TABLES',
        ];
    }
}
```